src/main/java/sa/assertj/iterables/AnyMatch.java

```diff
@@ -0,0 +1,34 @@
+package sa.assertj.iterables;
+
+import sa.assertj.Experiment;
+
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.function.Predicate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AnyMatch extends Experiment {
+    static Experiment.DataProvider provider = (size, numOfSamples) -> {
+        Object[][] result = new Object[numOfSamples][];
+        for (int i = 0; i < numOfSamples; i++) {
+            ArrayList<Integer> list = new ArrayList<>();
+            Random rand = new Random();
+            int element = rand.nextInt();
+            for (int j = 0; j < size; j++) {
+                list.add(element);
+            }
+
+            Predicate<Integer> predicate = s -> s.equals(element);
```

Comment:
So this one is the same as the `AllMatch` generator, right? I think that's actually not ideal, because it will always return after the first element (since that one will match). I think we should do something like this, where we afterwards choose the index we will be checking randomly:

```suggestion
            ArrayList<Integer> list = new ArrayList<>();
            Random rand = new Random();
            for (int j = 0; j < size; j++) {
                list.add(random.nextInt());
            }

            int index = rand.nextInt(size);
            Predicate<Integer> predicate = s -> s.equals(list.get(index));
```